integration_tests/src/main/python/hash_aggregate_test.py

```diff
@@ -883,6 +884,176 @@ def test_hash_groupby_collect_partial_replace_with_distinct_fallback(data_gen,
     group by a""",
         conf=conf)
 
+
+exact_percentile_data_gen = [ByteGen(), ShortGen(), IntegerGen(), LongGen(), FloatGen(), DoubleGen(),
+                             RepeatSeqGen(ByteGen(), length=100),
+                             RepeatSeqGen(ShortGen(), length=100),
+                             RepeatSeqGen(IntegerGen(), length=100),
+                             RepeatSeqGen(LongGen(), length=100),
+                             RepeatSeqGen(FloatGen(), length=100),
+                             RepeatSeqGen(DoubleGen(), length=100),
+                             FloatGen().with_special_case(math.nan, 500.0)
+                             .with_special_case(math.inf, 500.0),
+                             DoubleGen().with_special_case(math.nan, 500.0)
+                             .with_special_case(math.inf, 500.0)]
+
+exact_percentile_reduction_data_gen = [
+    [('val', data_gen),
+     ('freq', LongGen(min_val=0, max_val=1000000, nullable=False)
+                     .with_special_case(0, weight=100))]
+    for data_gen in exact_percentile_data_gen]
+
+def exact_percentile_reduction(df):
+    # df = spark.read.parquet("/home/nghiat/TMP/df.parquet")
+    # df.coalesce(1).write.mode("overwrite").parquet("/home/nghiat/TMP/df.parquet")
```

Comment:
nit: can we please remove the commented out code here?

Reply:
Ah sorry, forgot to do so.